Translation of units.py to java

Illinoia opened the pull request.

Illinoia committed:
Made some stuff

Hal-9k1 committed:
Add package statement, fix for loop syntax

Hal-9k1 commented:
- Try using Map and HashMap instead of Dictionary and Hashtable, as both of the later are outdated.
- We can't use a main method in our robot code; the robot controller has its own main method that
  will eventually call into our code through OpModes. To set up the static maps you have created,
  use a [static initializer block](https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html).
    - Make sure the maps are defined outside of the static initializer block.
- Try making a Unit enum inside the Units class and use that instead of strings to represent units.
  If you do this, you also don't have to check that the unit parameters are valid in the convert
  method because Java won't let users of your function use a nonexistant enum member.

bidenloh commented:
damn thats alot of importing stuff just for coding a robot

Commit:
Implement changes in PR comments

Use Distance and Angle enums to specify units instead of a string. This
is better type safety and removes the need to check arguments in convert.
Remove conversion tables and just put units per reference unit as an
enum field.
Multiple by to / from in conversion instead of from / to.
